src/controller/CHIPCluster.h

```diff
@@ -366,7 +366,7 @@ class DLL_EXPORT ClusterBase
             }
         };
 
-        auto onSubscriptionEstablishedCb = [context, subscriptionEstablishedCb]() {
+        auto onSubscriptionEstablishedCb = [context, subscriptionEstablishedCb](const app::ReadClient & readClient) {
```

Comment:
The readClient is unused in the scope. This should not fail to compile

Reply:
Maybe Controller::SubscribeAttribute requires a specific type of callback even if unused?

Either way, I would want to see the compiler error this gets. I don't see any compiler errors in CI.

Reply:
> Maybe Controller::SubscribeAttribute requires a specific type of callback even if unused?

The callback function signature has to be right, yes.  The function will be called with certain arguments; whether it uses those arguments is not something the caller has a way to know.

>  I don't see any compiler errors in CI.

That's because this is (1) a template and (2) not called by anything in CI, so not instantiated, so there are no "does it compile?" checks, since those generally would depend on the types the template is instantiated with.